enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,303 @@
+
+The bridge binding is able to expose the pod ip to the VM as expected by hypershift but kubevirt does not support live migration with it, for kubevirt to support live migration with bridge binding 
+at primary interface the OVN kubernetes CNI need to do the following:
+- not set the IP at pod netns
+- not expect the IP to be on the pod netns
+
+
+### Workflow Description
+
+The gist of the design is:
+
+1. Hypershift kubevirt workers attach to ovn-k node switches but CNI does not set IP addresses in virt-launcher pods netns
+2. DHCP is configured at VMs logical switch ports to deliver the address to VMs
+3. A point to point routing is used so one node's subnet ip can be routed from different node
+3. The VM's gateway ip and mac are independent of the node they are running on using proxy arp
+4. Current OVN arp proxy feature is missing following features: configure MAC and IPv6, they are being implemented [here](https://github.com/ovn-org/ovn/pull/176)
+
+
+### API Extensions
+
+n/a
+
+## Design Details
+
+### OVN-Kubernetes 
+
+#### Topology
+
+**Point to point routing:**
+
+Do point to point routing with a policy for outbound traffic and a static route for
+inbound, with this the VM can live migrate to different node and keep previous 
+address keeping n/s and e/w communication.
+
+
+```text
+       ┌───────────────────────────┐
+       │     ovn_cluster_router    │
+┌───────────────────────────┐   ┌────────────────────────────┐
+│ static route              │───│ policy                     │
+│ prefix: 10.244.0.8        │   │ match: ip4.src==10.244.0.8 │
+│ nexthop: 10.244.0.8       │   │ action: reroute            │
+│ output-port: rtos-node1   │   │ nexthop: 10.64.0.2         │
+│ policy: dst-ip            │   │                            │  
+└───────────────────────────┘   └────────────────────────────┘ 
+```
+**Nodes logical switch ports:**
+
+The "router" logical switch port per node need to activate "arp_proxy" option
+with the VMs gateway and a fixed mac so VM neighbor cache is consistent 
+after live migration
+
+```text
+┌────────────────────┐     ┌────────────────────┐
+│logical switch node1│     │logical switch node2│
+┌────────────────────┐  │     │   ┌─────────────────────┐      
+│ lsp stor-node1     │──┘     └───│ lsp stor-node2      │      
+│ options:           │            │ options:            │    
+│  arp_proxy:        │            │   arp_proxy:        │
+│   0a:58:0a:f3:00:00│            │    0a:58:0a:f3:00:00│
+│   169.254.1.1      │            │    169.254.1.1      │
+└────────────────────┘            └─────────────────────┘
+```
+
+**VMs logical switch ports:**
+
+The logical switch port for new VMs will use ovn-k ipam to reserve an IP address,
+and when live-migrating the VM, the LSP address will be re-used.
+
+CNI must avoid setting the IP address on the migration destination pod.
```

Comment:
But OVNK itself should still keep this IP allocated to the new pod, even though CNI wont configure it..right? Would be good to explicitly say that here.